src/System.Runtime.Intrinsics.X86/ref/System.Runtime.Intrinsics.X86.cs

```diff
@@ -0,0 +1,1122 @@
+        public static Vector256<ushort> SubtractSaturate(Vector256<ushort> left, Vector256<ushort> right) { throw null; }
+        public static Vector256<ulong> SumAbsoluteDifferences(Vector256<byte> left, Vector256<byte> right) { throw null; }
+        public static Vector256<sbyte> UnpackHigh(Vector256<sbyte> left, Vector256<sbyte> right) { throw null; }
+        public static Vector256<byte> UnpackHigh(Vector256<byte> left, Vector256<byte> right) { throw null; }
+        public static Vector256<short> UnpackHigh(Vector256<short> left, Vector256<short> right) { throw null; }
+        public static Vector256<ushort> UnpackHigh(Vector256<ushort> left, Vector256<ushort> right) { throw null; }
+        public static Vector256<int> UnpackHigh(Vector256<int> left, Vector256<int> right) { throw null; }
+        public static Vector256<uint> UnpackHigh(Vector256<uint> left, Vector256<uint> right) { throw null; }
+        public static Vector256<long> UnpackHigh(Vector256<long> left, Vector256<long> right) { throw null; }
+        public static Vector256<ulong> UnpackHigh(Vector256<ulong> left, Vector256<ulong> right) { throw null; }
+        public static Vector256<sbyte> UnpackLow(Vector256<sbyte> left, Vector256<sbyte> right) { throw null; }
+        public static Vector256<byte> UnpackLow(Vector256<byte> left, Vector256<byte> right) { throw null; }
+        public static Vector256<short> UnpackLow(Vector256<short> left, Vector256<short> right) { throw null; }
+        public static Vector256<ushort> UnpackLow(Vector256<ushort> left, Vector256<ushort> right) { throw null; }
+        public static Vector256<int> UnpackLow(Vector256<int> left, Vector256<int> right) { throw null; }
+        public static Vector256<uint> UnpackLow(Vector256<uint> left, Vector256<uint> right) { throw null; }
+        public static Vector256<long> UnpackLow(Vector256<long> left, Vector256<long> right) { throw null; }
+        public static Vector256<ulong> UnpackLow(Vector256<ulong> left, Vector256<ulong> right) { throw null; }
+        public static Vector256<sbyte> Xor(Vector256<sbyte> left, Vector256<sbyte> right) { throw null; }
+        public static Vector256<byte> Xor(Vector256<byte> left, Vector256<byte> right) { throw null; }
+        public static Vector256<short> Xor(Vector256<short> left, Vector256<short> right) { throw null; }
+        public static Vector256<ushort> Xor(Vector256<ushort> left, Vector256<ushort> right) { throw null; }
+        public static Vector256<int> Xor(Vector256<int> left, Vector256<int> right) { throw null; }
+        public static Vector256<uint> Xor(Vector256<uint> left, Vector256<uint> right) { throw null; }
+        public static Vector256<long> Xor(Vector256<long> left, Vector256<long> right) { throw null; }
+        public static Vector256<ulong> Xor(Vector256<ulong> left, Vector256<ulong> right) { throw null; }
+    }
+    public static class Bmi1
+  
```

Comment:
@russellhadley @terrajobst @CarolEidt @mellinoe: Could we get a formal decision on whether having the `SSE` and `SSE2` instructions in a single class or split into their respective classes is the way to go here.

There was some discussion on the issue thread, but it didn't really go anywhere and there wasn't any discussion over it in the design meeting.

My thoughts are that we should be providing separate `Sse` and `Sse2` classes and not combining them together, regardless of whether the current runtime requires both `SSE` and `SSE2` support for x86 and x64.

I believe that this provides the most consistent API and doesn't leave future questions as to how to expose various APIs (as might come up with future platforms/architectures, such as ARM).

Reply:
I think we should separate them into two classes.